### PR TITLE
Update tab panel tabs to look more like tabs

### PR DESCRIPTION
**Pull request by 2metres**
**User-Facing Changes**
Update tab panel tabs to look more like tabs in light mode

**Description**
Screen Shot 2022-05-10 at 11 52 24 am

### CODE REVIEW

File: packages/studio-base/src/components/EmptyPanelLayout.tsx [hunk 1]
```diff
@@ -28,22 +26,18 @@ type Props = {
...
-  },
-  dropzoneOver: {
+const Root = muiStyled("div")(({ theme }) => ({
+  backgroundColor: theme.palette.background.paper,
+  width: "100%",
+  height: "100%",
+  overflowY: "auto",
+}));
+
+const DropTarget = muiStyled("div")<{ isOver: boolean }>(({ isOver, theme }) => ({
```
Comment (defunctzombie) [inline packages/studio-base/src/components/EmptyPanelLayout.tsx:36]: Whats the thinking for preferring muiStyled over useStyles?

File: packages/studio-base/src/panels/Tab/DraggableToolbarTab.tsx [hunk 2]
```diff
@@ -45,10 +40,18 @@ export function DraggableToolbarTab(props: Props): JSX.Element {
...
-  const [{ isOver }, dropRef] = useDrop<TabLocation, void, { isOver: boolean }>({
+  const [{ highlight }, dropRef] = useDrop<
+    TabLocation,
+    void,
+    { highlight: "before" | "after" | undefined }
+  >({
     accept: TAB_DRAG_TYPE,
-    collect: (monitor) => ({
-      isOver: monitor.isOver(),
+    collect: (monitor: DropTargetMonitor<TabLocation, void>) => ({
```
Comment (defunctzombie) [inline packages/studio-base/src/panels/Tab/DraggableToolbarTab.tsx:49]: Maybe this should be extracted to a separate function? I find the layers of nested ternary hard to read and reason about. Comments might help too.

File: packages/studio-base/src/panels/Tab/index.tsx [hunk 3]
```diff
@@ -30,13 +30,13 @@ import { SaveConfig } from "@foxglove/studio-base/types/panels";
 import { TAB_PANEL_TYPE } from "@foxglove/studio-base/util/globalConstants";
 import { DEFAULT_TAB_PANEL_CONFIG, updateTabPanelLayout } from "@foxglove/studio-base/util/layout";
 
-const SPanelCover = styled.div`
+const SPanelCover = muiStyled("div")`
```
Comment (defunctzombie) [inline packages/studio-base/src/panels/Tab/index.tsx:33]: Maybe separate component files help organize and keep files more focused?

**Review by defunctzombie — APPROVED** [review]
Looks nicer. Left some thoughts on code structure. Something we don't do so well today but I'd like to try towards is keeping 1 component per component file. I think that will help us re-use and structure our components in a cleaner way.

File: packages/studio-base/src/panels/Tab/DraggableToolbarTab.tsx (hunk 2 above)
Comment (2metres) [reply packages/studio-base/src/panels/Tab/DraggableToolbarTab.tsx:49]: @jtbandes can you weigh in here?

File: packages/studio-base/src/components/EmptyPanelLayout.tsx (hunk 1 above)
Comment (2metres) [reply packages/studio-base/src/components/EmptyPanelLayout.tsx:36]: `makeStyles` has been marked as legacy by MUI. `muiStyled` solves the styling issue easily without introducing the sx prop which we all agree has performance issues. By using `muiStyled` here I reduced the style approaches in this file alone from 3 to 1.

**Comment by 2metres** [comment]
I think the idea of 1 component per file is not neccisarily susutainable. Especially in the world of styled-components

**Comment by amacneil** [comment]
👍 big improvement

@2metres it could use a tiny bit more styling for the inactive tabs - right now they are just floating with no borders or hover state.

image

**Pull request merged**